src/lib/asl.c

```diff
@@ -0,0 +1,56 @@
+#include <asl.h>
+#include <pwd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <time.h>
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+#include <xhyve/asl.h>
+
+static aslclient asl = NULL;
+static aslmsg log_msg = NULL;
+
+static unsigned char *buf = NULL;
+static size_t buf_size = 0;
+static size_t buf_capacity = 0;
+
+/* asl is deprecated in favor of os_log starting with macOS 10.12.  */
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+/* Grow buf/buf_capacity. */
+static void buf_grow(void)
+{
+	buf_capacity = buf_capacity ? 2 * buf_capacity : 1024;
+	buf = realloc(buf, buf_capacity);
+	if (!buf) {
+		perror("buf_grow");
+		exit(1);
+	}
```

Comment:
I'd have used `xhyve_abort` but this is ok too I guess.

Reply:
There were several `exit(1)` already.  I don't know which coding style to follow, there are too many.

Reply:
I think it depends on whether the code was originally kernel, library or application code in the FreeBSD bhyve stack. Looks like `xhyve_abort` is only in "kernel" code so that was a bad suggestion, sorry. In src/lib `abort` and `exit` are basically even so lets stick with `exit`.